src/workerd/server/server.c++

```diff
@@ -1566,7 +1570,40 @@ private:
 
   kj::Promise<void> writeLogfwdr(uint channel,
       kj::FunctionParam<void(capnp::AnyPointer::Builder)> buildMessage) override {
-    KJ_FAIL_REQUIRE("no logging channels");
+    auto& context = IoContext::current();
+
+    auto headers = kj::HttpHeaders(context.getHeaderTable());
+    auto client = context.getHttpClient(channel, true, nullptr, "writeLogfwdr"_kjc);
+
+    auto urlStr = kj::str("https://fake-host");
+ 
+    capnp::MallocMessageBuilder requestMessage;
+    auto requestBuilder = requestMessage.initRoot<capnp::AnyPointer>();
+  
+    buildMessage(requestBuilder);
+    capnp::JsonCodec json;
+    auto requestJson = json.encode(requestBuilder.getAs<api::AnalyticsEngineEvent>());
+
+    return context.waitForOutputLocks().then([client = kj::mv(client), urlStr = kj::mv(urlStr),
+        requestJson = kj::mv(requestJson), headers = kj::mv(headers)]() mutable {
+      auto innerReq = client->request(kj::HttpMethod::POST, urlStr, headers, requestJson.size());
+
+      struct RefcountedWrapper: public kj::Refcounted {
+        explicit RefcountedWrapper(kj::Own<kj::HttpClient> client): client(kj::mv(client)) {}
+        kj::Own<kj::HttpClient> client;
+      };
+      auto rcClient = kj::refcounted<RefcountedWrapper>(kj::mv(client));
+      auto req = attachToRequest(kj::mv(innerReq), kj::mv(rcClient));
+
```

Comment:
This should be updated to use coroutine (co_await/co_return) syntax rather than using kj::Promise `.then` syntax.